pkg/store/proxy.go

```diff
@@ -12,41 +12,37 @@ import (
 	"github.com/pkg/errors"
 	"github.com/prometheus/tsdb/labels"
 	"golang.org/x/sync/errgroup"
-	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 // Info holds meta information about a store.
-type Info struct {
-	Addr string
-
+type Client interface {
 	// Client to access the store.
-	Client storepb.StoreClient
+	storepb.StoreClient
 
 	// Labels that apply to all date exposed by the backing store.
-	Labels []storepb.Label
+	Labels() []storepb.Label
 
 	// Minimum and maximum time range of data in the store.
-	MinTime, MaxTime int64
-}
+	RangeTime() (mint int64, maxt int64)
 
-func (i *Info) String() string {
-	return i.Addr
+	// String representation for logging mostly.
+	String() string
```

Comment:
Do we ever call this actually? Logging, `fmt`, and friends will infer a `fmt.Stringer` automatically.

Reply:
Nope, I don't think so